cadc-rest/src/main/java/ca/nrc/cadc/rest/RestServlet.java

```diff
@@ -484,25 +484,62 @@ static void setAuthenticateHeaders(Subject subject, String realm, SyncOutput out
             
         } else {
             log.debug("adding challenges for " + im.getClass().getName());
+            RegistryClient rc = new RegistryClient();
+            LocalAuthority localAuthority = new LocalAuthority();
             for (URI sm :  im.getSecurityMethods()) {
                 String challenge = SEC_METHOD_CHALLENGES.get(sm);
                 log.debug(sm + " -> " + challenge);
                 if (challenge != null) {
-                    // TODO: check System.getProperty(CERT_HEADER_ENABLE) aka trust ingress to do client cert validation?
-                    // TODO: check for duplicate challenges (map does contain them)?
                     StringBuilder sb = new StringBuilder();
                     sb.append(challenge);
+
+                    // try to add alternative challenges that are augmented with info on how to acquire such a credential
+                    // catch and log failures at debug so this cannot break things
                     if (challenge.equals(AuthenticationUtil.CHALLENGE_TYPE_BASIC)) {
+                        // augment the primary challenge
                         sb.append(" realm=\"").append(realm).append("\"");
+                    } else if (challenge.equals(AuthenticationUtil.CHALLENGE_TYPE_IVOA_X509)
+                            || challenge.equals(AuthenticationUtil.CHALLENGE_TYPE_IVOA_COOKIE)) {
+                        final URI standardID;
+                        if (challenge.equals(AuthenticationUtil.CHALLENGE_TYPE_IVOA_X509)) {
+                            standardID = Standards.CRED_PROXY_10;
+                        } else {
+                            // make something up for now
+                            standardID = URI.create("ivo://ivoa.net/std/Cred#cookie-issuer");
+                        }
+                        try {
+                            URI certGenService = localAuthority.getServiceURI(standardID.toASCIIString());
```

Comment:
`credGenService` instead of `certGenService` in anticipation of the cookie-issuer service coming online?
Relatedly, I don't think lines in the block below (512 down) will be correct if it does come online.  So there's only a partial step towards a working cookie issuing service.  Just need to remember this code will need updating.